tasks/actorstate/miner.go

```diff
@@ -132,23 +129,58 @@ func NewMinerStateExtractionContext(ctx context.Context, a ActorInfo, node Actor
 		CurrActor: &a.Actor,
 		CurrState: curState,
 		CurrTs:    a.TipSet,
+		Address:   a.Address,
+		Store:     node.Store(),
+		API:       node,
+		cache:     NewDiffCache(),
```

Comment:
This should allow us to keep nearly the same performance we had before the refactor. This caches the results of Diffs for a single extraction.